### PR TITLE
3987 - fixes docstrings and adds checks

### LINKED ISSUE #3987: Dictionary version of Monai Transpose
Hello all. In the task I am working on, I am incorrectly reading the volume images with CacheDataset, because the shape they have is HxWxD. When creating the loader, I want shape (B,C,D,H,W), and I am getting (B,C,H,W,D). I saw the transformation transpose, that should be enough passing the correct indexes as arguments. Nevertheless, I am dealing with a dictionary in my loaders, with keys ['image'] and ['label']. Would it be possible to have a "Transposed" in the docs?
Thank you for your time

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #3987

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_module_list.py [hunk 1]
```diff
@@ -37,17 +38,29 @@ def test_public_api(self):
...
         xforms = {
             name: obj
             for name, obj in monai.transforms.__dict__.items()
             if inspect.isclass(obj) and issubclass(obj, monai.transforms.MapTransform)
         }
         names = sorted(x for x in xforms if x not in to_exclude)
         remained = set(names)
+        doc_file = os.path.join(pathlib.Path(__file__).parent.parent, "docs", "source", "transforms.rst")
+        with open(doc_file) as f:
+            contents = f.readlines()
```
Comment (ericspod) [inline tests/test_module_list.py:52]: `contents = f.read()` then don't need the next line?

**Review by ericspod — APPROVED** [review]

File: tests/test_module_list.py [hunk 2]
```diff
@@ -37,17 +38,29 @@ def test_public_api(self):
...
+        contents = "".join(contents)
+
         for n in names:
             if not n.endswith("d"):
                 continue
             basename = n[:-1]  # Transformd basename is Transform
+            for docname in (f"{basename}", f"{basename}d"):
+                if docname in to_exclude_docs:
+                    continue
+                if f"`{docname}`" not in contents:
```
[the remark this answers is not in the document]
Comment (wyli) [reply tests/test_module_list.py]: thanks, the subtests work well, I also tried `self.assertIn` but the error message includes the entire `contents` which is tedious here.

**Pull request merged**